### PR TITLE
Distribution contract documentation

### PR DESCRIPTION
**Pull request by foxpy**

### CODE REVIEW

File: docs/neutron/distribution/messages.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+```rust
+pub enum ExecuteMsg {
+    TransferOwnership(String),
+    SetShares {
+        shares: Vec<(String, Uint128)>,
+    },
+    Fund {},
+    Claim {},
+}
```
Comment (albertandrejev) [inline docs/neutron/distribution/messages.md:11]: Please add `Pause`/`Unpause` messages descriptions.

File: docs/neutron/distribution/queries.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Queries
+
+```rust
+pub enum QueryMsg {
+    Config {},
+    Pending {},
+    Shares {},
+}
```
Comment (albertandrejev) [inline docs/neutron/distribution/queries.md:8]: Please add `PauseInfo` description with return type.

File: docs/neutron/distribution/messages.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Messages
+
+```rust
+pub enum ExecuteMsg {
+    TransferOwnership(String),
```
Comment (NeverHappened) [inline docs/neutron/distribution/messages.md:5]: how about putting comments here as well - like we with Albert did? It's kinda easy to read then

File: docs/neutron/distribution/overview.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Overview
+
+This document describes the Distribution contract for the Neutron network.
+
+Distribution contract is instantiated in genesis and owned by DAO. It fullfils the purpose of _distributing_ earned
```
Comment (NeverHappened) [inline docs/neutron/distribution/overview.md:5]: typo -> `fullfills`

File: docs/neutron/distribution/queries.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+```
+
+## Config
+
+Returns current config of contract. Config has a following schema:
+
+```rust
+pub struct Config {
+    pub denom: String,
```
Comment (NeverHappened) [inline docs/neutron/distribution/queries.md:17]: No comment for denom

File: docs/neutron/distribution/overview.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+tokens between configurable set of shareholders.
+
+This contract stores shares as an array of `(address, weight)` pairs and distributes incoming funds among
+shareholders by their weight.
+
+Contract can only be configured by DAO. DAO can do two things to this contract:
+- transfer ownership to anyone else;
+- alter shareholders' weights.
```
Comment (NeverHappened) [inline docs/neutron/distribution/overview.md:13]: `'` non in place

File: docs/neutron/distribution/messages.md (hunk 3 above)
Comment (NeverHappened) [reply docs/neutron/distribution/messages.md:5]: and also how about putting permissions in these comments as well like we did? [Example](https://github.com/neutron-org/neutron-docs/pull/41/files#diff-bb04acdf8088b51282bc195b4e5cdd78809f8291b8d7160c787d74e6e4e2ffe0R28)

**Review by NeverHappened — CHANGES_REQUESTED** [review]
Wrote about small things.

Also debatable point:
I feel like this contract needs more clarification about it's mechanics -> how `fund`ing works, why do we have this contract (not sure, maybe it's apparent)

File: docs/neutron/distribution/overview.md (hunk 6 above)
Comment (foxpy) [reply docs/neutron/distribution/overview.md:13]: Please refer to guidline §3 here: https://guides.lib.uoguelph.ca/c.php?g=696372&p=4999533

File: docs/neutron/distribution/queries.md (hunk 5 above)
Comment (foxpy) [reply docs/neutron/distribution/queries.md:17]: It is a direct copy paste from source code. But I can add a little clarification comment :+1:

File: docs/neutron/distribution/overview.md (hunk 4 above)
Comment (foxpy) [reply docs/neutron/distribution/overview.md:5]: Turns out we are both wrong! https://www.grammarly.com/blog/fulfil-fulfill/

File: docs/neutron/distribution/messages.md (hunk 3 above)
Comment (foxpy) [reply docs/neutron/distribution/messages.md:21]: Made it more uniform

**Comment by foxpy** [comment]
> I feel like this contract needs more clarification about it's mechanics

done :+1: 

> why do we have this contract

I decided to avoid this part

**Review by NeverHappened — APPROVED** [review]

**Review by albertandrejev — APPROVED** [review]

**Pull request merged**